pkg/records/payer_record.go

```diff
@@ -0,0 +1,214 @@
+
+	// Required. If the Transfer Agent Indicator in position 133 is a
+	// “1” (one), enter the city, town, or post office of the transfer
+	// agent. Otherwise, enter payer’s city, town, or post office city.
+	// Do not enter state and ZIP Code information in this field. Left
+	// justify the information and fill unused positions with blanks.
+	PayerCity string `json:"payer_city" validate:"required"`
+
+	// Required. Enter the valid U.S. Postal Service state abbreviation.
+	PayerState string `json:"payer_state" validate:"required"`
+
+	// Required. Enter the valid nine-digit ZIP Code assigned by the
+	// U.S. Postal Service. If only the first five digits are known, left
+	// justify the information and fill unused positions with blanks. For
+	// foreign countries, alpha characters are acceptable as long as
+	// the filer has entered a “1” (one) in “A” Record, field position 52
+	// Foreign Entity Indicator.
+	PayerZipCode string `json:"payer_zip_code" validate:"required"`
+
+	// Enter the payer’s telephone number and extension. Omit
+	// hyphens. Left justify the information and fill unused positions
+	// with blanks.
+	PayerTelephoneNumber string `json:"payer_telephone_number_and_ext"`
+
+	// Required. Enter the number of the record as it appears within
+	// the file. The record sequence number for the “T” Record will
+	// always be “1” (one), since it is the first record on the file and
+	// the file can have only one “T” Record. Each record thereafter
+	// must be increased by one in ascending numerical sequence,
+	// that is, 2, 3, 4, etc. Right justify numbers with leading zeros in
+	// the field. For example, the “T” Record sequence number
+	// would appear as “00000001” in the field, the first “A” Record
+	// would be “00000002,” the first “B” Record, “00000003,” the
+	// second “B” Record, “00000004” and so on until the final record
+	// of the file, the “F” Record.
+	RecordSequenceNumber int `json:"record_sequence_number" validate:"required"`
+}
+
+// Type returns type of “A” record
+func (r *ARecord) Type() string {
+	return config.ARecordType
+}
+
+// Parse parses the “A” record from fire ascii
+func (r *ARecord) Parse(buf []byte) error {
+	record := string(buf)
+	if utf8.RuneCountInString(record) != config.RecordLength {
+		return utils.ErrRecordLength
+	}
+
+	fields := reflect.ValueOf(r).Elem()
+	if !fields.IsValid() {
+		return utils.ErrValidField
+	}
+
+	return utils.ParseValue(fields, config.ARecordLayout, record)
+}
+
+// Ascii returns fire ascii of “A” record
+func (r *ARecord) Ascii() []byte {
+	var buf strings.Builder
+	records := config.ToSpecifications(config.ARecordLayout)
+	fields := reflect.ValueOf(r).Elem()
+	if !fields.IsValid() {
+		return nil
+	}
+
+	buf.Grow(config.RecordLength)
+	for _, spec := range records {
+		value := utils.ToString(spec.Field, fields.FieldByName(spec.Name))
+		buf.WriteString(value)
+	}
+
+	return []byte(buf.String())
```

Comment:
We could use a `bytes.Buffer` here instead. That has `WriteString(string)` and can `return buf.Bytes()` at the end. It would save the extra type conversion.

Reply:
fixed